crates/autopilot/src/infra/persistence/auction/dto.rs

```diff
@@ -0,0 +1,309 @@
+use {
+    crate::{
+        boundary::{self},
+        domain,
+    },
+    number::serialization::HexOrDecimalU256,
+    primitive_types::{H160, U256},
+    serde::{Deserialize, Serialize},
+    serde_with::serde_as,
+    std::collections::BTreeMap,
+};
+
+pub fn from_domain(auction: domain::Auction) -> Auction {
+    Auction {
+        block: auction.block,
+        latest_settlement_block: auction.latest_settlement_block,
+        orders: auction.orders.into_iter().map(Into::into).collect(),
+        prices: auction.prices,
+    }
+}
+
+pub fn to_domain(auction: Auction) -> domain::Auction {
+    domain::Auction {
+        block: auction.block,
+        latest_settlement_block: auction.latest_settlement_block,
+        orders: auction.orders.into_iter().map(Into::into).collect(),
+        prices: auction.prices,
+    }
+}
+
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Auction {
+    pub block: u64,
+    pub latest_settlement_block: u64,
+    pub orders: Vec<Order>,
+    #[serde_as(as = "BTreeMap<_, HexOrDecimalU256>")]
+    pub prices: BTreeMap<H160, U256>,
+}
+
+pub type AuctionId = i64;
+
+#[serde_as]
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AuctionWithId {
+    pub id: AuctionId,
+    #[serde(flatten)]
+    pub auction: Auction,
+}
+
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Order {
+    pub uid: boundary::OrderUid,
+    pub sell_token: H160,
+    pub buy_token: H160,
+    #[serde_as(as = "HexOrDecimalU256")]
+    pub sell_amount: U256,
+    #[serde_as(as = "HexOrDecimalU256")]
+    pub buy_amount: U256,
+    #[serde_as(as = "HexOrDecimalU256")]
+    pub solver_fee: U256,
+    #[serde_as(as = "HexOrDecimalU256")]
+    pub user_fee: U256,
+    pub valid_to: u32,
+    pub kind: boundary::OrderKind,
+    pub receiver: Option<H160>,
+    pub owner: H160,
+    pub partially_fillable: bool,
+    #[serde_as(as = "HexOrDecimalU256")]
+    pub executed: U256,
+    pub pre_interactions: Vec<boundary::InteractionData>,
+    pub post_interactions: Vec<boundary::InteractionData>,
+    pub sell_token_balance: boundary::SellTokenSource,
+    pub buy_token_balance: boundary::BuyTokenDestination,
+    pub class: boundary::OrderClass,
+    pub app_data: boundary::AppDataHash,
+    #[serde(flatten)]
+    pub signature: boundary::Signature,
+}
+
+impl From<domain::Order> for Order {
```

Comment:
Can we be consistent on whether we use Into trait implementation of "to_domain()"/"from_domain()" methods? 

I don't remember the exact reason, but I remember that people preferred explicit functions (ie `to_domain()`) over trait implementation unless it was explicitly needed.

Reply:
True, not sure but to_domain/from_domain methods seem clearer to me at this point as whenever I see that I know that some abstraction layer change is happening beneath (maybe just a habit). Anyway, switched for consistency.